test/integration/test_api_logging.py

```diff
@@ -0,0 +1,337 @@
+        format=f"{TEST_FORMAT} %(levelname)s %(message)s",
+        level=log_level,
+        force=True,
+    )
+
+
+def reset_logging():
+    logging.basicConfig(
+        format="%(asctime)s %(levelname)-8s %(message)s",
+        level=logging.ERROR,
+        datefmt="%Y-%m-%d %H:%M:%S",
+        force=True,
+    )
+    loggerDict = logging.root.manager.loggerDict
+    for key in list(loggerDict.keys()):
+        loggerDict[key].disabled = True
+        del loggerDict[key]
+
+
+@pytest.fixture
+def custom_logging():
+    @contextlib.contextmanager
+    def logging_context_creator(log_level: int):
+        configure_logging(log_level)
+        yield
+        reset_logging()
+
+    return logging_context_creator
+
+
+@pytest.fixture
+def luigi_output(tmp_path):
+    set_build_config(
+        False,
+        tuple(),
+        False,
+        False,
+        str(tmp_path),
+        str(tmp_path.parent),
+        "",
+        "test",
+    )
+    return tmp_path
+
+
+@ignore_resource_warning()
+def dummy_api_command(log_level: Optional[str], use_job_specific_log_file: bool):
+    task_creator = lambda: generate_root_task(task_class=DummyTask)
+    result = run_task(
+        task_creator=task_creator,
+        workers=2,
+        task_dependencies_dot_file=None,
+        log_level=log_level,
+        use_job_specific_log_file=use_job_specific_log_file,
+    )
+    return result
+
+
+def assert_loggers_are_equal(logger_infos_after, logger_infos_before):
+    assert logger_infos_before[ROOT_LOGGER] == logger_infos_after[ROOT_LOGGER]
+    assert (
+        logger_infos_before[API_CLIENT_LOGGING_TEST_LOGGER]
+        == logger_infos_after[API_CLIENT_LOGGING_TEST_LOGGER]
+    )
+    assert logger_infos_before[LUIGI_LOGGER] == logger_infos_after[LUIGI_LOGGER]
+    assert (
+        logger_infos_before[LUIGI_INTERFACE_LOGGER]
+        == logger_infos_after[LUIGI_INTERFACE_LOGGER]
+    )
+
+
+def create_test_regex(log_level: int):
+    level_name = logging.getLevelName(log_level)
+    regex = f".*{TEST_FORMAT} {level_name} DummyTask_.* DUMMY LOGGER {level_name}.*"
+    return re.compile(regex)
+
+
+def test_luigi_log_level_info_and_basic_logging_error(
+    capfd, custom_logging, luigi_output
+):
+    """
+    This test checks if setting luigi log level to level INFO and global log level to ERROR
+    prints error/info messages to stderr.
+    Note: This test does not work with `pytest -s ...`
+    """
+
+    with custom_logging(log_level=logging.ERROR) as error_log:
+        logger_infos_before = create_logger_infos()
+        result = dummy_api_command(log_level="INFO", use_job_specific_log_file=False)
+        logger_infos_after = create_logger_infos()
+        assert_loggers_are_equal(logger_infos_after, logger_infos_before)
+
+        stdout_output, stderr_output = capfd.readouterr()
+        assert stdout_output == ""
+        assert stderr_output != ""
+        assert create_test_regex(logging.ERROR).search(stderr_output) is not None
```

Comment:
we maybe should use the [regex_matcher](https://github.com/exasol/integration-test-docker-environment/blob/0ca00b38c9199970a4d5b380da67f5d5f8745522/test/matchers.py#L4), it produces nicer output